astropy/nddata/nddata.py

```diff
@@ -220,6 +222,13 @@ def __init__(self, data, uncertainty=None, mask=None, wcs=None,
             # than sorry :-)
             unit = deepcopy(unit)
 
+        # Validate the wcs
+        if isinstance(wcs, BaseLowLevelWCS):
```

Comment:
I just realised (via this breaking my code) that this logic is backwards, we need to check for highlevel first, and only wrap if it's not highlevel but is low level.